nexus/src/app/update.rs

```diff
@@ -23,18 +30,27 @@ impl super::Nexus {
         body: impl Stream<Item = Result<Bytes, HttpError>> + Send + Sync + 'static,
         file_name: String,
```

Comment:
Maybe we could take this as a follow-up, but I'm going to put in a vote for removing this `file_name` parameter and all of it associated copies. It seems confusing and that it's not really doing much for us.

Reply:
#8560